zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/instance/ElementInstance.java

```diff
@@ -52,6 +52,15 @@ public final class ElementInstance extends UnpackedObject implements DbValue {
   private final ObjectProperty<TaskListenerIndicesRecord> taskListenerIndicesRecordProp =
       new ObjectProperty<>("taskListenerIndicesRecord", new TaskListenerIndicesRecord());
 
+  /**
+   * This value is added in 8.7, any child process instances created before 8.7 will have a depth of
+   * 0.
+   *
+   * @since 8.7
```

Comment:
🔧 Child process instances created after 8.7, but as part of root process instance that was created prior to 8.7 would not have a correct depth. Their depth would be counted as the depth since 8.7, i.e. the first child process instance since 8.7 will have depth `1`.

🔧 We should document this point in the description of the configuration option as well for correctness.